File: Javascript/Algorithms/1287.js
/**
 * @param {number[]} arr
 * @return {number}
 */
let findSpecialInteger = arr => {
    const length = arr.length;
    let threshold = Math.ceil(length * 0.25), count = 1;
    for (let i = 1; i < length; i++){
        if (arr[i-1] == arr[i]){
            count++;
            if (count > threshold)
                return arr[i];
        } else 
            count = 1;
    }
    return arr[0];
};

File: C/Algorithms/1287.c
int findSpecialInteger(int *arr, int arrSize) {
  int threshold = (arrSize * 0.25), count = 1;
  for (int i = 1; i < arrSize; i++) {
    if (*(arr + i - 1) == *(arr + i)) {
      count++;
      if (count > threshold)
        return *(arr + i);
    } else
      count = 1;
  }
  return arr[0];
}
